src/routes/courierPage.tsx

```diff
@@ -1,314 +1,314 @@
-  ]);
-  const [selectedSize, setSelectedSize] =
-    useState<SingleValue<{ value: string; label: string }>>(null);
-  const [availableCompartments, setAvailableCompartments] = useState<
-    CompartmentOption[]
-  >([]);
-  const [selectedCompartment, setSelectedCompartment] = useState<string>("");
-  const [selectedCourier, setSelectedCourier] = useState<string>("Amazon");
-
-  const db = getFirestore();
-
-  // Fetch users from Firestore
-  useEffect(() => {
-    const fetchUsers = async () => {
-      const usersCollection = collection(db, "users");
-      const userDocs = await getDocs(usersCollection);
-      const userList = userDocs.docs.map((doc) => ({
-        value: doc.id,
-        label: doc.data().username,
-      }));
-      setUsers(userList);
-    };
-
-    fetchUsers();
-  }, [db]);
-
-  // Fetch compartments when a size is selected
-  useEffect(() => {
-    if (!selectedSize) return;
-
-    const fetchCompartments = async () => {
-      try {
-        const vaultsCollection = collection(db, "vault");
-        const compartmentsQuery = query(
-          vaultsCollection,
-          where("packsize", "==", selectedSize.value),
-          where("status", "==", "available")
-        );
-        const compartmentDocs = await getDocs(compartmentsQuery);
-
-        const compartments = compartmentDocs.docs.map((doc) => ({
-          id: doc.id,
-          compartment: doc.data().compartment,
-        }));
-
-        setAvailableCompartments(compartments);
-      } catch (error) {
-        Swal.fire({
-          icon: "error",
-          title: "Error",
-          text: "Failed to fetch compartments. Please try again later.",
-        });
-      }
-    };
-
-    fetchCompartments();
-  }, [selectedSize, db]);
-
-  const handleSubmit = async () => {
-    console.log("Selected User:", selectedUser);
-    console.log("Selected Size:", selectedSize);
-    console.log("Selected Compartment:", selectedCompartment);
-    console.log("Selected Courier:", selectedCourier);
-
-    if (
-      !selectedUser ||
-      !selectedSize ||
-      !selectedCompartment ||
-      !selectedCourier
-    ) {
-      Swal.fire({
-        icon: "error",
-        title: "Missing Fields",
-        text: "Please fill in all the fields before submitting!",
-      });
-      return;
-    }
-
-    if (availableCompartments.length === 0) {
-      Swal.fire({
-        icon: "error",
-        title: "No Compartments Available",
-        text: "Please select a valid compartment size with available compartments.",
-      });
-      return;
-    }
-
-    try {
-      const deliveryCollection = collection(db, "delivery");
-      const vaultCollection = collection(db, "vault");
-
-      // Query the vault collection for the selected compartment
-      const vaultQuery = query(
-        vaultCollection,
-        where("compartment", "==", isNaN(Number(selectedCompartment))
-          ? selectedCompartment // Treat as string if not a valid number
```

Comment:
Be careful with this assumption here, this value is coming from the user. And it sent to your database, it might allow them to inject malicious code in your db.